### PR TITLE
added folder component

### PR DESCRIPTION
**Pull request by 3rfan**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #issuenum
| Related issues/PRs | #issuenum
| License | MIT

#### What's in this PR?

The folder component

#### Why?

The Datagrid needs a folder view.

#### Example Usage

~~~js
// Single folder
<Folder
    id="1"
    info="3 Objects"
    title="This is a folder"
    onClick={handleClick}
/>

// Folder list
<FolderList>
    <FolderList.Folder
        id="1"
        info="3 Objects"
        title="This is a folder"
    />
    <FolderList.Folder
        id="2"
        info="2 Objects"
        title="This is a folder"
    />
    <FolderList.Folder
        id="3"
        info="0 Objects"
        title="This is a folder"
    />
</FolderList>
~~~

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/Folder.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import Icon from '../Icon';
+import folderStyles from './folder.scss';
+
+const FOLDER_ICON = 'folder-o';
+
+type Props = {
+    /** The id which will be passed as an argument inside the onClick handler */
+    id: string | number,
+    /** The subtext underneath the title */
+    meta: string,
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/Folder.js]: Hm... `meta` is a bit a strange name... Doesn't really say anything. Maybe `description`? Or `info`?

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/Folder.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    render() {
+        const {
+            meta,
+            title,
+        } = this.props;
+
+        return (
+            <div
+                className={folderStyles.folder}
+                onClick={this.handleClick}
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/Folder.js]: That's not very accessible. What about including a `a` or `button` tag somehow?

Comment (danrot) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/Folder.js]: I've just realized that the same is true for our `Icon` component 🙈

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+The FolderList component is used display a list of folders. The FolderList accepts Folder componenta as children.
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/README.md]: `used to display`

Comment (danrot) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/README.md]: `Folder components`

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/README.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+The Folder component is just an interactive element for presentation purposes. Every Folder component has its
+own onClick-handler. The FolderList component offers a more convenient way to work with multiple folders by offering 
+a single onClick-handler.
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/README.md]: Please use backticks if you are referencing code, like for `onClick`, `Folder` or `FolderList`

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/README.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
+The FolderList component is used display a list of folders. The FolderList accepts Folder componenta as children.
+The Folder component is just an interactive element for presentation purposes. Every Folder component has its
+own onClick-handler. The FolderList component offers a more convenient way to work with multiple folders by offering
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/README.md]: Maybe split the sentences here? After `has its own onClick-handler` you could bring the `Folder` example, and `The FolderList component offers a more...` could be the text before the `FolderList`example.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/Folder.js (hunk 2 above)
Comment (3rfan) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/Folder.js]: I am not sure what you mean. I have added `role="button"` and `tabIndex="0"` in the next 2 lines for accessibility. The reason is that you can't use divs inside a button.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/README.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
+The `FolderList` component is used to display a list of folders. The `FolderList` accepts `Folder` components as 
+children. The `Folder` component is just an interactive element for presentation purposes. Every `Folder` component has 
+its own `onClick`-handler.
+
+`Folder` example:
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/README.md]: I would delete that line, we have that nowhere else.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/Folder.js (hunk 2 above)
Comment (danrot) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/FolderList/Folder.js]: ah, looks like I have overseen that 🙈

**Pull request merged**